frontend/src/DataImporter/buildNetwork.ts

```diff
@@ -0,0 +1,223 @@
+import {EdgeDataRow, NodeDataRow, EdgeSpectrumDataRow} from "./parseCsv";
+
+export interface Edge {
+  id: string;
+  node1Id: string;
+  node2Id: string;
+  totalCapacity: string;
+  provisionedCapacity: number;
+}
+
+export interface Node {
+  id: string;
+  latitude: number;
+  longitude: number;
+  neighbors: {
+    node: Node;
+    edge: Edge;
+  }[];
+}
+
+export interface Chanel {
+  id: string;
+  chanel_label: string;
+  nodes: string[];
+  frequency: number;
+  width: number;
+  // wavelength: number;
+}
+export interface ChanelEdge {
+  id: string;
+  chanel_label: string;
+  edges: string[];
+  frequency: number;
+  width: number;
+  // wavelength: number;
+}
+
+export interface Network {
+  nodes: Node[];
+  edges: Edge[];
+  chanels: Chanel[];
+}
+
+export const handleEdge = (edgeData: EdgeDataRow, nodes: Node[]): Edge => {
+  const node1 = nodes.find(node => node.id === edgeData.node1);
+  const node2 = nodes.find(node => node.id === edgeData.node2);
+
+  if (!node1 || !node2) {
+    throw new Error(`Edge ${edgeData.id} references non-existing nodes`);
+  }
+
+
+  const edge: Edge = {
+    id: edgeData.id,
+    node1Id: node1.id,
+    node2Id: node2.id,
+    totalCapacity: edgeData.totalCapacity,
+    provisionedCapacity: edgeData.provisionedCapacity,
+  };
+
+  node1.neighbors.push({node: node2, edge});
+  node2.neighbors.push({node: node1, edge});
+
+  return edge;
+}
+
+export const mergeEdges = (edges: EdgeDataRow[]): EdgeDataRow[] =>{
+    let merged: EdgeDataRow[] = []
+    edges.map(
+      edge => {
+        // finding corresponding edge
+        const pair = edges.find(element => element.node1 == edge.node2 && element.node2 == edge.node1)
+        // check if corresponding edge exists
+        if(typeof pair !== 'undefined' ){
+          // check if corresponding edge has been written in merged - push if not - ignore if yes
+          if(typeof merged.find(element => element.id == pair.id) === 'undefined'){
+            merged.push(edge)
+          }
+        }
+        else{
+          throw new Error(`Can't merge: ${JSON.stringify(edge)} has no pair`);
+        }
+      }
+    )
+    return merged;
+}
+
+export const handleNode = (nodesData: NodeDataRow[]): Node[] =>{
+  return nodesData.map(nodeData => ({
+    id: nodeData.id,
+    latitude: nodeData.latitude,
+    longitude: nodeData.longitude,
+    neighbors: [],
+  }));
+}
+
+export const checkEdgeExists = (chanelData: EdgeSpectrumDataRow[], edges: EdgeDataRow[]): boolean =>{
+  const edgeIDs: string[] = edges.map(element => element.id)
+  chanelData.map(chanel => {
+    if(!edgeIDs.includes(chanel.edgeId)){
+      throw new Error(`Edge does not exists: ${JSON.stringify(chanel)} edge id does not apper in EdgeDataRow`);
+    }
+  })
+  return true
+}
+
+export const getChanel = (chanelData: EdgeSpectrumDataRow, chanels: ChanelEdge[]): ChanelEdge[] => {
+  const cur_id = chanelData.channelId;
+  const found = chanels.find((chanel) => chanel.id == cur_id)
```

Comment:
to nie jest nigdzie używane?

Reply:
chodzi o zmienną found

Reply:
found jest sprawdzane linijka niżej.
Jest po to by sprawdzić czy dany kanał został wprowadzony na listę kanałów. Na tej podstawie albo dopisuje się do niego kolejną krawędź, albo dodaje się nowy kanał do listy kanałów.